app/api/probe_dock/metrics_api.rb

```diff
@@ -117,6 +119,51 @@ def current_organization
         result
       end
 
+      get :projectHealth do
+        authorize! :organization, :data
+
+        test_descriptions_rel = TestDescription.joins(:project_version)
+
+        # Retrieve the data from the specified version otherwise take the most recent project version to retrieve the data
+        if params[:projectVersionId].present?
+          project_version = ProjectVersion.where('api_id = ?', params[:projectVersionId]).first
+          test_descriptions_rel = test_descriptions_rel.joins(:project_version).where('project_versions.api_id = ?', params[:projectVersionId])
+        else
+          project_version = ProjectVersion.joins(:project).where('projects.api_id = ?', params[:projectId]).order('created_at DESC').first
+          test_descriptions_rel = test_descriptions_rel.where('project_versions.id = ?', project_version.id)
```

Comment:
If you change the condition to `test_descriptions.project_version_id = ?`, you can remove the join at line 125. This will also avoid a duplicate join in the `if`.